vite.config.ts

```diff
@@ -0,0 +1,50 @@
+import { configDefaults, defineConfig, UserConfig } from "vitest/config";
+import react from "@vitejs/plugin-react";
+import svgr from "vite-plugin-svgr";
+import { loadEnv } from "vite";
+import fs from "node:fs";
+import path from "node:path";
+
+// Support optionally pulling in external branding if the module is installed.
+const theme = "@microbit-foundation/python-editor-v3-microbit";
+const external = `node_modules/${theme}`;
+const internal = "src/deployment/default";
+
+export default defineConfig(({ mode }) => {
+  const commonEnv = loadEnv(mode, process.cwd(), "");
+  const unitTest: UserConfig["test"] = {
+    globals: true,
+    exclude: [...configDefaults.exclude, "**/e2e/**"],
+    environment: "jsdom",
+    setupFiles: "./src/setupTests.ts",
+  };
+  const e2eTest: UserConfig["test"] = {
+    globals: true,
+    include: ["src/e2e/**/*.test.ts"],
+    testTimeout: 60_000,
+    hookTimeout: 30_000,
+  };
+  const config: UserConfig = {
+    base: process.env.BASE_URL ?? "/",
+    build: {
+      outDir: "build",
+    },
+    worker: {
+      format: "es",
```

Comment:
Is this an issue for browser support? Might break Safari 14 which otherwise has OK support: https://caniuse.com/mdn-api_worker_worker_ecmascript_modules

Reply:
Confirmed this breaks search in Safari 14.1 as the worker fails to initialize.

Reply:
It works in Safari 15 even on iOS.

With type: "module" in the worker init Vite will still try to bundle to an iife by default.
However, that doesn't work if the worker itself dynamically loads code.
We do that for the stemmers as there's a blob of code/data per language. There's not an obvious way forward here.

One approach could be to disable search on platforms where the worker init fails. We already disable search in for languages that don't have Lunr support.

Perhaps it's possible to build a worker per language instead? That might be quite a change. Though I suppose it could be simple if verbose to have a worker file per language with the relevant import. Would need app changes to cope as previously the language change was internal to the worker.

Reply:
> Perhaps it's possible to build a worker per language instead

Did this. A bit messy but it should be quicker to init as we remove a waterfall step. 
Looks good on Safari 14.